src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/ElfObjectWriter.cs

```diff
@@ -141,6 +141,7 @@ private protected override void CreateSection(ObjectNodeSection section, string
                 {
                     Type = type,
                     Flags = flags,
+                    Alignment = section.Type == SectionType.Debug ? 1 : 0
```

Comment:
```suggestion
                    Alignment = 1
```

I think we can skip this check. Presumably any section will get some alignment set through `UpdateSectionAlignment` based on the alignment requirement of the data inside.